### ISSUE TITLE
Keep compared files open after closing Differ

### ISSUE THREAD

**Issue by pintassilgo** [CudaText-addons/cuda_differ#22]
In both Notepad++ and Kate, my previous editors, I'm used to still be able to edit files after I end a diff.

It's not that this is impossible to do in Differ, but I don't find it practical: I need to right click the tab, hover Differ then click "Back to separate tabs".

It would be good if the files can keep open after I just close the Differ tab.

The proposal made by @Alexey-T in #20 to not close original tabs when opening Differ, at the cost of removing the ability to save edits made in Differ tab, was rejected by users who need to save those edits. One possible workaround: when an edit in Differ tab is saved, the original tab-file should be updated to reflect the change, i.e. keep original tab-file synced with the tab-file in `differ_backup` while Differ is running.

Other proposal is what I said in #20, I believe much easier to implement and without needing to worry about side effects: create the `differ.separate_on_close` option, defaulting to `false` to not change behavior for current users. If `true`, closing a differ tab will automatically restore tabs, which is the same as the running the command "Back to separate tabs".

**Comment by Alexey-T** [comment]
>I wrote a possible solution above (sync original file with the one in differ_backup), does it sound reasonable?

maybe not: it is not simple to sync them (how to do it? just copy contents of file-tag in Differ-tab to the original file?)

**Comment by Alexey-T** [comment]
These are ok idea, can you help with coding it?
@ildarkhasanshin - are u interested to help?

**Comment by Alexey-T** [comment]
Cud dont allow to open the same stored filename in 2 tabs. And Differ does this - it opens the original filenames in the differ pair.

**Comment by Alexey-T** [comment]
Can you code this change, please.

**Comment by ildarkhasanshin** [comment]
@pintassilgo
1 rename /py/cuda_differ to /py/cuda_differ.bak
2 download https://github.com/ildarkhasanshin/cuda_differ/archive/refs/heads/master.zip
3 copy updated cuda_differ to /py/
4 test it

**Comment by Alexey-T** [comment]
Better rename "cuda_duffer-master" to valid Python name. It must be valid ID.

**Comment by ildarkhasanshin** [comment]
> Better rename "cuda_differ-master" to valid Python name. It must be valid ID.

github itself gives such a name for the zip-link

**Comment by CudaText-addons** [comment]
btw, I only look at this topic talks, but I don't work on it. maybe @pintassilgo you can make your own change.

**Comment by ildarkhasanshin** [comment]
@CudaText-addons, мы ведь не можем закрыть untitled-табы через без предупреждения?

**Comment by ildarkhasanshin** [comment]
> Back to separate tabs command, which I expected to run automatically when I close the differ-tab, doesn't have these issues in release version (no delay and no duplicated tabs).

no, `untitled`-tabs still doesn't close

**Comment by ildarkhasanshin** [comment]
> Comparing two unsaved (untitled) files: original tabs don't close when I run Differ. When I close differ-tab, the files created in differ_backup folder are reopened (with delay). so now I have duplicated tabs.

fixed
@pintassilgo, https://github.com/CudaText-addons/cuda_differ/issues/22#issuecomment-2748968251

**Comment by ildarkhasanshin** [comment]
> ~2 seconds delay between closing the differ-tab

in code 0.3 sec

**Comment by ildarkhasanshin** [comment]
@pintassilgo, are you sure you are using this last version `https://github.com/ildarkhasanshin/cuda_differ` ?

https://github.com/user-attachments/assets/e44b4e75-6cb0-4f52-98ea-f12e8f21e02a

**Comment by ildarkhasanshin** [comment]
fixed
@pintassilgo, https://github.com/CudaText-addons/cuda_differ/issues/22#issuecomment-2748968251

**Comment by ildarkhasanshin** [comment]
> I ask to compare Untitled28 and Untitled29. Then I close the differ-tab. When the tabs are restored, they are restored as Untitled30 and 31.

@Alexey-T, это, скорее всего, зависит от текущей сессии ct?

**Comment by ildarkhasanshin** [comment]
@Alexey-T, возможно после закрытия untitled-вкладки не увеличивать порядковый номер?

**Comment by CudaText-addons** [comment]
@ildarkhasanshin 
нет, но можно прочитать и установить заголовки табов, PROP_TAB_TITLE.
